packages/react/src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.jsx

```diff
@@ -258,9 +258,14 @@ const DataSeriesFormItem = ({
       ? cardConfig?.content?.series
       : cardConfig?.content?.attributes;
 
-  // initialize items with a unique id if not present
+  // initialize items with a unique dataSourceId if not present
   const dataSection = useMemo(
-    () => data?.map((item) => (!item.uuid ? { ...item, uuid: uuid.v4() } : item)),
+    () =>
+      data?.map((item) =>
+        item.dataSourceId?.length > 36 // TODO: check if there is uuid present in the dataSourceId string
```

Comment:
could we just compare the dataSourceId to the dataItem name to determine this?  if they're the same, then we should generate the UUID

Reply:
Good point. I will update this